src/python/botan3.py

```diff
@@ -1771,14 +1813,144 @@ def _load_buf_or_file(filename, buf, file_fn, buf_fn):
+    def handle_(self):
+        return self.__obj
+
+    def set_common_name(self, name):
+        _DLL.botan_x509_cert_opts_common_name(self.__obj, _ctype_str(name))
+
+    def setcountry(self, country):
+        _DLL.botan_x509_cert_opts_country(self.__obj, _ctype_str(country))
+
+    def set_organization(self, organization):
+        _DLL.botan_x509_cert_opts_organization(self.__obj, _ctype_str(organization))
+
+    def set_org_unit(self, org_unit):
+        _DLL.botan_x509_cert_opts_org_unit(self.__obj, _ctype_str(org_unit))
+
+    def set_locality(self, locality):
+        _DLL.botan_x509_cert_opts_locality(self.__obj, _ctype_str(locality))
+
+    def set_state(self, state):
+        _DLL.botan_x509_cert_opts_state(self.__obj, _ctype_str(state))
+
+    def set_serial_number(self, serial_number):
+        _DLL.botan_x509_cert_opts_serial_number(self.__obj, _ctype_str(serial_number))
+
+    def set_email(self, email):
+        _DLL.botan_x509_cert_optsemail(self.__obj, _ctype_str(email))
+
+    def set_uri(self, uri):
+        _DLL.botan_x509_cert_opts_uri(self.__obj, _ctype_str(uri))
+
+    def set_ip(self, ip):
+        _DLL.botan_x509_cert_opts_ip(self.__obj, _ctype_str(ip))
+
+    def set_dns(self, dns):
+        _DLL.botan_x509_cert_opts_dns(self.__obj, _ctype_str(dns))
+
+    def set_xmpp(self, xmpp):
+        _DLL.botan_x509_cert_opts_xmpp(self.__obj, _ctype_str(xmpp))
+
+    def set_challenge(self, challenge):
+        _DLL.botan_x509_cert_opts_challenge(self.__obj, _ctype_str(challenge))
+
+    def set_more_org_units(self, more_org_units):
+        cnt = len(more_org_units)
+        _DLL.botan_x509_cert_opts_more_org_units(self.__obj, (c_char_p * cnt)(*(_ctype_str(x) for x in more_org_units)), c_size_t(cnt))
+
+    def set_more_dns(self, more_dns):
+        cnt = len(more_dns)
+        _DLL.botan_x509_cert_opts_more_dns(self.__obj, (c_char_p * cnt)(*(_ctype_str(x) for x in more_dns)), c_size_t(cnt))
+
+    def set_ca_key(self, limit):
+        _DLL.botan_x509_cert_opts_ca_key(self.__obj, c_size_t(limit))
+
+    def set_padding_scheme(self, scheme):
+        _DLL.botan_x509_cert_opts_set_padding_scheme(self.__obj, _ctype_str(scheme))
+
+    def set_not_before(self, not_before):
+        _DLL.botan_x509_cert_opts_not_before(self.__obj, not_before.handle_())
+
+    def set_not_after(self, not_after):
+        _DLL.botan_x509_cert_opts_not_after(self.__obj, not_after.handle_())
+
+    def set_constraints(self, usage_list):
+        # TODO this sucks, where enum
+        usage_values = {"NO_CONSTRAINTS": 0,
+                        "DIGITAL_SIGNATURE": 32768,
+                        "NON_REPUDIATION": 16384,
+                        "KEY_ENCIPHERMENT": 8192,
+                        "DATA_ENCIPHERMENT": 4096,
+                        "KEY_AGREEMENT": 2048,
+                        "KEY_CERT_SIGN": 1024,
+                        "CRL_SIGN": 512,
+                        "ENCIPHER_ONLY": 256,
+                        "DECIPHER_ONLY": 128}
```

Comment:
This exists in 3 different places now, I really wish it were an enum. But I suppose that would be a breaking change for `X509Cert().allowed_usage()`?

Reply:
Also apart from the above, this version is a setter, so it might be better to yell at the user when they pass something invalid rather than silently `pass`